geomstats/geometry/discrete_curves.py

```diff
@@ -1577,12 +1458,30 @@ def space_derivative(curve):
         mat_3 = from_vector_to_diagonal_matrix(vec_3, 1)
         mat_space_deriv = mat_1 - mat_2 + mat_3
 
-        space_deriv = n_points * gs.matmul(mat_space_deriv, curve)
+        return n_points * gs.matmul(mat_space_deriv, curve)
+
+
+class SRVPreShapeSpace(DiscreteCurves):
```

Comment:
If I understand correctly, this is supposed to represent the base space ? If so, the name is a bit misleading because the base space is usually called the shape space, equipped with the quotient metric, while the total space is called the pre-shape space, and equipped with e.g. the SRV metric.

Reply:
More generally, I think it would be more natural if the shape space had the same structure as the space of curves: only one class `DiscreteShapes` or `UnparametrizedDiscreteCurves` (the equivalent of `DiscreteCurves`) and several quotient metrics, among which one (e.g. `QuotientSRVMetric`) is chosen as the default metric, just like `SRVMetric` is the default metric of discrete curves. What do you think @luisfpereira ?